Source/WebCore/platform/graphics/skia/GraphicsContextSkia.cpp

```diff
@@ -895,12 +898,9 @@ void GraphicsContextSkia::drawPattern(NativeImage& nativeImage, const FloatRect&
     if (spacing.isZero() && !needsClip)
         paint.setShader(image->makeShader(SkTileMode::kRepeat, SkTileMode::kRepeat, samplingOptions, &shaderMatrix));
     else {
-        if (needsClip) {
-            // FIXME: handle the case where the tile rect has a different size than the image.
-            notImplemented();
-        }
         SkPictureRecorder recorder;
-        auto* recordCanvas = recorder.beginRecording(SkRect::MakeWH(tileRect.width() + spacing.width() / patternTransform.a(), tileRect.height() + spacing.height() / patternTransform.d()));
+        auto* recordCanvas = recorder.beginRecording(tileRect.width() + spacing.width() / patternTransform.a(), tileRect.height() + spacing.height() / patternTransform.d());
```

Comment:
What is this changing?

Reply:
that's some leftover refactoring I tried - it's worth to cleanup such cases in separate PR I guess